### ISSUE TITLE
Bitbucket event source: no active event server running

### ISSUE THREAD

**Issue by guysarig** [argoproj/argo-events#1549]
I'm trying to install bitbucket EventSource at my argo events installation. After applying the EventSource yaml, a service + pod are being created (as expected, as happens at other EventSources I used), but the pod (the newly-created bitbucket eventsource pod) is getting into CrashLoopBackOff state after a couple of secs, with "event source: no active event server running" message in it's log.

I'm using Kubernetes 1.19.13 over EKS, with a out-of-the-box (no customizations made) installation of Argo events 1.5.5 - namespace installation (tried upgrading to 1.5.6 - no change), made according to the [official docs](https://argoproj.github.io/argo-events/installation/).

This is the yaml that i'd applied for creating the bitbucket EventSource (replaced the names with some dummy names):
```
apiVersion: argoproj.io/v1alpha1
kind: EventSource
metadata:
  name: bitbucket
spec:
  service:
    ports:
      - port: 12000
        targetPort: 12000
  bitbucket:
    # bitbucket eventsource example with oauth bearer token auth strategy
    example-oauth-token:
      # key of the project
      projectKey: "SOMEKEY"
      # owner of the repository
      owner: "test"
      active: "true"
      # repository slug
      repositorySlug: "somerepo"
      # Bitbucket will send events to following port and endpoint
      webhook:
        # endpoint to listen to events on
        endpoint: /push
        # port to run internal HTTP server on
        port: "12000"
        # HTTP request method to allow. In this case, only POST requests are accepted
        method: POST
        # url the event-source will use to register in Bitbucket.
        # This url must be reachable from outside the cluster.
        # The name for the service is in `<event-source-name>-eventsource-svc` format.
        # You will need to create an Ingress or Openshift Route for the event-source service so that it can be reached from Bitbucket.
        url: https://argo-events.myorg.com
      # events to listen to
      # Visit https://support.atlassian.com/bitbucket-cloud/docs/manage-webhooks/
      events:
        - "*"
      auth:
        # oauthToken refers to K8s secret that stores the bitbucket oauth bearer token
        # Visit https://support.atlassian.com/bitbucket-cloud/docs/use-oauth-on-bitbucket-cloud/
        oauthToken:
          # Name of the K8s secret that contains the access token
          name: bitbucket-access
          # Key within the K8s secret whose corresponding value (must be base64 encoded) is access token
          key: token
      # Delete the webhook when the eventsource is deleted
      deleteHookOnFinish: true
```

(also tried applying the example yaml from [argo-events repo](https://github.com/argoproj/argo-events/blob/master/examples/event-sources/bitbucket.yaml) - no luck)

This is my pod 's completed log (I toggled DEBUG_LOG flag for more verbosity) :
```
2022-01-25T23:28:33.689Z	INFO	argo-events.eventsource	cmd/start.go:63	starting eventsource server	{"eventSourceName": "bitbucket", "version": "v1.5.6"}
2022-01-25T23:28:33.689Z	INFO	argo-events.eventsource	eventsources/eventing.go:318	Starting event source server...	{"eventSourceName": "bitbucket"}
2022-01-25T23:28:33.689Z	INFO	argo-events.eventsource	metrics/metrics.go:172	starting metrics server	{"eventSourceName": "bitbucket"}
2022-01-25T23:28:33.689Z	INFO	argo-events.eventsource	driver/nats.go:93	NATS auth strategy: Token	{"eventSourceName": "bitbucket", "clientID": "client-bitbucket-eventsource-2qc5g-df75b97b7-gbqvt-166"}
2022-01-25T23:28:33.692Z	INFO	argo-events.eventsource	driver/nats.go:105	Connected to NATS server.	{"eventSourceName": "bitbucket", "clientID": "client-bitbucket-eventsource-2qc5g-df75b97b7-gbqvt-166"}
2022-01-25T23:28:33.693Z	INFO	argo-events.eventsource	driver/nats.go:118	Connected to NATS streaming server.	{"eventSourceName": "bitbucket", "clientID": "client-bitbucket-eventsource-2qc5g-df75b97b7-gbqvt-166"}
2022-01-25T23:28:33.693Z	INFO	argo-events.eventsource	eventsources/eventing.go:438	Eventing server started.	{"eventSourceName": "bitbucket"}
2022-01-25T23:28:33.693Z	ERROR	argo-events.eventsource	eventsources/eventing.go:455	Erroring out, no active event server running	{"eventSourceName": "bitbucket"}
github.com/argoproj/argo-events/eventsources.(*EventSourceAdaptor).run
	/home/runner/work/argo-events/argo-events/eventsources/eventing.go:455
github.com/argoproj/argo-events/eventsources.(*EventSourceAdaptor).Start
	/home/runner/work/argo-events/argo-events/eventsources/eventing.go:293
github.com/argoproj/argo-events/eventsources/cmd.Start
	/home/runner/work/argo-events/argo-events/eventsources/cmd/start.go:65
github.com/argoproj/argo-events/cmd/commands.NewEventSourceCommand.func1
	/home/runner/work/argo-events/argo-events/cmd/commands/eventsource.go:14
github.com/spf13/cobra.(*Command).execute
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960
github.com/spf13/cobra.(*Command).Execute
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
github.com/argoproj/argo-events/cmd/commands.Execute
	/home/runner/work/argo-events/argo-events/cmd/commands/root.go:19
main.main
	/home/runner/work/argo-events/argo-events/cmd/main.go:8
runtime.main
	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/proc.go:255
2022-01-25T23:28:33.693Z	INFO	argo-events.eventsource	eventsources/eventing.go:341	starting eventbus connection daemon...	{"eventSourceName": "bitbucket"}
2022-01-25T23:28:33.693Z	INFO	argo-events.eventsource	eventsources/eventing.go:347	exiting eventbus connection daemon...	{"eventSourceName": "bitbucket"}
2022-01-25T23:28:33.694Z	ERROR	argo-events.eventsource	driver/nats.go:84	NATS connection lost	{"eventSourceName": "bitbucket", "clientID": "client-bitbucket-eventsource-2qc5g-df75b97b7-gbqvt-166"}
github.com/argoproj/argo-events/eventbus/driver.(*natsStreaming).Connect.func1
	/home/runner/work/argo-events/argo-events/eventbus/driver/nats.go:84
github.com/nats-io/nats%2ego.(*Conn).close.func1
	/home/runner/go/pkg/mod/github.com/nats-io/nats.go@v1.13.0/nats.go:4720
github.com/nats-io/nats%2ego.(*asyncCallbacksHandler).asyncCBDispatcher
	/home/runner/go/pkg/mod/github.com/nats-io/nats.go@v1.13.0/nats.go:2523
2022-01-25T23:28:33.694Z	FATAL	argo-events.eventsource	cmd/start.go:66	failed to start eventsource server	{"eventSourceName": "bitbucket", "error": "no active event server running", "errorVerbose": "no active event server running\ngithub.com/argoproj/argo-events/eventsources.(*EventSourceAdaptor).run\n\t/home/runner/work/argo-events/argo-events/eventsources/eventing.go:458\ngithub.com/argoproj/argo-events/eventsources.(*EventSourceAdaptor).Start\n\t/home/runner/work/argo-events/argo-events/eventsources/eventing.go:293\ngithub.com/argoproj/argo-events/eventsources/cmd.Start\n\t/home/runner/work/argo-events/argo-events/eventsources/cmd/start.go:65\ngithub.com/argoproj/argo-events/cmd/commands.NewEventSourceCommand.func1\n\t/home/runner/work/argo-events/argo-events/cmd/commands/eventsource.go:14\ngithub.com/spf13/cobra.(*Command).execute\n\t/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856\ngithub.com/spf13/cobra.(*Command).ExecuteC\n\t/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960\ngithub.com/spf13/cobra.(*Command).Execute\n\t/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897\ngithub.com/argoproj/argo-events/cmd/commands.Execute\n\t/home/runner/work/argo-events/argo-events/cmd/commands/root.go:19\nmain.main\n\t/home/runner/work/argo-events/argo-events/cmd/main.go:8\nruntime.main\n\t/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/proc.go:255\nruntime.goexit\n\t/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/asm_amd64.s:1581"}
github.com/argoproj/argo-events/eventsources/cmd.Start
	/home/runner/work/argo-events/argo-events/eventsources/cmd/start.go:66
github.com/argoproj/argo-events/cmd/commands.NewEventSourceCommand.func1
	/home/runner/work/argo-events/argo-events/cmd/commands/eventsource.go:14
github.com/spf13/cobra.(*Command).execute
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:856
github.com/spf13/cobra.(*Command).ExecuteC
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:960
github.com/spf13/cobra.(*Command).Execute
	/home/runner/go/pkg/mod/github.com/spf13/cobra@v1.1.3/command.go:897
github.com/argoproj/argo-events/cmd/commands.Execute
	/home/runner/work/argo-events/argo-events/cmd/commands/root.go:19
main.main
	/home/runner/work/argo-events/argo-events/cmd/main.go:8
runtime.main
	/opt/hostedtoolcache/go/1.17.1/x64/src/runtime/proc.go:255
```

**Environment**
 - Kubernetes: 1.19.13
 - Argo: 1.5.5
 - Argo Events: 1.5.5

---

**Message from the maintainers**:

If you wish to see this enhancement implemented please add a 👍 reaction to this issue! We often sort issues this way to know what to prioritize.

**Comment by daniel-codefresh** [comment]
Hey @guysarig , the recently added `bitbucket` event source (which integrates with the Bitbucket Cloud API) wasn't included in neither v1.5.5 release nor v1.5.6 release. ATM, it can only be used with `latest` tag. This explains the `no active event server running` errors you're getting because your version of Argo Events does not recognize this type of event source.
FYI, the older `bitbucketserver` event source which integrates with the Bitbucket Server API is already present in these releases.

**Comment by guysarig** [comment]
@daniel-codefresh, I missed that...
I pulled from master, will use latest until it'll be included in the stable version.

Thanks!
